src/main/java/net/dv8tion/jda/core/entities/Message.java

```diff
@@ -1254,4 +1262,247 @@ public boolean isImage()
+        {
+            JSONObject jsonObject = new JSONObject();
+            jsonObject.put("type", this.type.id).put("session_id", partyId);
+            return jsonObject;
+        }
+    }
+
+    /**
+     * Represents the {@link net.dv8tion.jda.core.entities.Message} application if the
+     * {@link net.dv8tion.jda.core.entities.Message.ActivityType ActivityType} of the
+     * {@link net.dv8tion.jda.core.entities.Message.Activity Activity} equals
+     * {@link net.dv8tion.jda.core.entities.Message.ActivityType#GAME ActivityType.GAME}.
+     *
+     * <br>All applications can be found at:
+     * <a href="https://discordapp.com/api/v7/games" target="_blank">https://discordapp.com/api/v7/games</a>.
+     */
+    class Application
+    {
+        private final String name;
+        private final String description;
+        private final String iconId;
+        private final String coverId;
+        private final long id;
+
+        public Application(String name, String description, String iconId, String coverId, long id)
+        {
+            this.name = name;
+            this.description = description;
+            this.iconId = iconId;
+            this.coverId = coverId;
+            this.id = id;
+        }
+
+        /**
+         *
+         * @return the applications name.
+         */
+        public String getName()
+        {
+            return name;
+        }
+
+        /**
+         *
+         * @return the applications description.
+         */
+        public String getDescription()
+        {
+            return description;
+        }
+
+        /**
+         *
+         * @return the applications icon id.
+         */
+        public String getIconId()
+        {
+            return iconId;
+        }
+
+        /**
+         *
+         * @return the applications cover image/id.
+         */
+        public String getCoverId()
+        {
+            return coverId;
+        }
+
+        /**
+         *
+         * @return the applications id.
+         */
+        public long getId()
+        {
+            return id;
+        }
+
+        /**
+         *
+         * @return {@link org.json.JSONObject} that can be used for sending an {@link net.dv8tion.jda.core.entities.Message.Activity Activity} in a {@link net.dv8tion.jda.core.entities.Message}.
+         */
+        public JSONObject toJSONObject()
+        {
+            JSONObject jsonObject = new JSONObject();
+            jsonObject.put("name", name).put("id", Long.toUnsignedString(id));
+            if (iconId != null)
+                jsonObject.put("icon", iconId);
+            if (description != null)
+                jsonObject.put("description", description);
+            if (coverId != null)
+                jsonObject.put("cover_image", coverId);
+            return jsonObject;
+        }
+    }
+
+    /**
+     * An enum containing {@link net.dv8tion.jda.core.entities.Message.Activity Activity} types
+     */
+    enum ActivityType
```

Comment:
We don't know enough about this yet, we should ask for better documentation from discord